core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,33 @@
+{% macro get_show_grant_sql(relation) %}
+{{ return(adapter.dispatch('get_show_grant_sql', 'dbt')(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+{{ return('show grants on table'(relation.schema)) }}
```

Comment:
Additionally, there's some post-processing logic that needs to happen during/after this query, to accomplish two things:
- Conform to a standardized format
- Ignore grants to the _current user_ (dbt's user), since we don't actually want to compare / revoke these! This this will include `ownership`, etc.

For the purposes of this PR, we need to figure out **what that standardized format needs to look like**. The information we need is: WHO (user/group) has WHICH PRIVILEGE on THIS RELATION, so that we can calculate diffs and template out the appropriate `revoke` / `grant` statements.

---

This is what `show grants` returns on Snowflake:
```
| created_on                     | privilege | granted_on | name                       | granted_to | grantee_name | grant_option | granted_by |
|--------------------------------|-----------|------------|----------------------------|------------|--------------|--------------|------------|
| 2022-06-02 01:52:28.935 -0700  | OWNERSHIP | TABLE      | AD_HOC.DBT_JCOHEN.MY_TABLE | ROLE       | AD_HOC       | true         | AD_HOC     |
```

This is what `show grants` returns on Databricks:
```
| Principal          | ActionType    | ObjectType   | ObjectKey           |
|--------------------|---------------|--------------|---------------------|
| jeremy@dbtlabs.com | OWN           | TABLE        | `airbnb`.`calendar` |
```

This is what the analogous query returns on Postgres:
```
grantor | grantee | table_catalog | table_schema | table_name | privilege_type | is_grantable | with_hierarchy
---------+---------+---------------+--------------+------------+----------------+--------------+----------------
 jerco   | other   | jerco         | dbt_jcohen   | whatever   | INSERT         | NO           | NO
 jerco   | other   | jerco         | dbt_jcohen   | whatever   | SELECT         | NO           | YES
 jerco   | another | jerco         | dbt_jcohen   | whatever   | INSERT         | NO           | NO
 jerco   | another | jerco         | dbt_jcohen   | whatever   | SELECT         | NO           | YES
```

And on BigQuery:
```
| object_catalog | object_schema | object_name | object_type | privilege_type            | grantee                  |
|----------------|---------------|-------------|-------------|---------------------------|--------------------------|
| dbt-test-env   | dbt_jcohen    | my_table    | TABLE       | roles/bigquery.dataViewer | user:jeremy@example.com  |
| dbt-test-env   | dbt_jcohen    | my_table    | TABLE       | roles/custom.whatEver     | user:another@example.com |
```

([The Redshift version of this query will be much messier](https://stackoverflow.com/questions/18741334/how-do-i-view-grants-on-redshift), unfortunately)

---

The nice thing on Postgres/Redshift/BigQuery is that, because it's a real `select` query, we can do all that processing right in the SQL query: just the columns we need, renamed to conformed standard, with current user's grants filtered out. For databases that support `show grants`, we need to do that post-processing in Jinja/Python instead. If that processing proves very difficult, Snowflake does give us the option of querying an `information_schema` view instead (https://docs.snowflake.com/en/sql-reference/info-schema/object_privileges.html). There are some performance costs to using `information_schema` instead of `show` on Snowflake, which we can discuss if/when we get there.